### PR TITLE
add centerLeading to AppBar.

### PR DESCRIPTION
**Pull request by ghost**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue. An issue is not required if the PR fixes something trivial like a typo.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by nate-thegrate — COMMENTED** [review]
Hi @mu7ammadkamel, thanks for contributing!

I'd like to highlight one of the items in the "pre-launch checklist":

> - [ ] I listed at least one issue that this PR fixes in the description above.

I think opening a [feature request](https://github.com/flutter/flutter/issues/new?assignees=&labels=&projects=&template=3_feature_request.yml) would be super helpful here, since it would allow us to learn about the intended use case and decide whether the change is worth pursuing.
(A single extra class field isn't a huge deal, but the API becomes slightly more cumbersome each time we add something.)

**Comment by nate-thegrate** [comment]
> There is already an option to center the leading widget, but it only works with widgets of the IconButton type. If I use a rounded widget that isn't an IconButton, it won't be centered.

True that!

Is it worth adding a `centerLeading` flag to match `centerTitle` when the alternative (wrapping with a `Center` widget) is equally trivial? I'm not sure ¯\\\_(ツ)\_/¯

File: packages/flutter/test/material/app_bar_test.dart [hunk 1]
```diff
@@ -13,7 +13,8 @@ import 'app_bar_utils.dart';
 
 TextStyle? _iconStyle(WidgetTester tester, IconData icon) {
   final RichText iconRichText = tester.widget<RichText>(
-    find.descendant(of: find.byIcon(icon).first, matching: find.byType(RichText)),
+    find.descendant(
+        of: find.byIcon(icon).first, matching: find.byType(RichText)),
```
Comment (justinmc) [inline packages/flutter/test/material/app_bar_test.dart:17]: Can you remove all of this autoformatting?

**Review by justinmc — COMMENTED** [review]
+1 to @nate-thegrate's points. Can't you achieve the same thing by wrapping your `leading` parameter with `Center`? Consider filling out the PR description and filing an issue with some more background info.

**Comment by Piinks** [comment]
(PR Triage): Hey @mu7ammadkamel thank you for contributing, would you like to continue working on this change? It sounds like this is something we don't want to add at this time, perhaps you can provide more context for the change? Thanks!

**Pull request closed without merging**